src/index.js

```diff
@@ -0,0 +1,165 @@
+                    action: '/lessons/new',
+                  },
+                  {
+                    text: instructorsTitleText,
+                    action: '/instructors',
+                  },
+                  {
+                    text: guideTitleText,
+                    action: guideUrl,
+                  },
+                  {
+                    text: chatTitleText,
+                    action: chatUrl,
+                  },
+                  {
+                    text: logOutTitleText,
+                    action: logout,
+                  },
+                ]}
+              />
+
+              <Main>
+
+                <Switch>
+
+                  <Route 
+                    exact
+                    path='/'
+                    render={() => (
+                      <Request url={data.instructor_url}>
+                        {({data}) => (
+                          <div>
+                            <Overview instructor={data} />
+                          </div>
+                        )}
+                      </Request>
+                    )}
+                  />
+
+                  <Route 
+                    exact
+                    path='/lessons/new'
+                    render={() => (
+                      <Request url={data.instructor_url}>
+                        {({data}) => (
+                          <New instructor={data} />
+                        )}
+                      </Request>
+                    )}
+                  />
+
+                  <Route 
+                    path={`/lessons/:slug`}
+                    render={({match}) => (
+                      <Request url={`/api/v1/lessons/${match.params.slug}`}>
+                        {({data}) => (
+                          <Lesson lesson={data} />
+                        )}
+                      </Request>
+                    )}
+                  />
+
+                  <Route 
+                    path={`/instructors/:slug`}
+                    render={({match}) => (
+                      <Request url={`/api/v1/instructors/${match.params.slug}`}>
+                        {({data}) => (
+                          <Overview instructor={data} />
+                        )}
+                      </Request>
+                    )}
+                  />
+
+                  <Route
+                    path='/instructors'
+                    render={() => (
+                      <Request url='/api/v1/instructors'>
+                        {({data}) => (
+                          <Instructors instructors={data} />
+                        )}
+                      </Request>
+                    )}
+                  />
+
+                  <Route component={RouteNotFound} />
+
+                </Switch>
+
+              </Main>
+
+            </div>
+          </BrowserRouter>
+        )
+      }}
+    </Request>
+  )
+}
+
+ReactDOM.render(
+  <App />,
+  document.getElementById('root')
+)
```

Comment:
This file is the entry point for the app; wires up authentication and routes - and it's a stateless functional component 💪